### PR TITLE
Add new CLI command to set BRANCHLIB

### PR DESCRIPTION
**Pull request by worksofliam**

### CODE REVIEW

File: cli/src/index.ts [hunk 1]
```diff
@@ -96,6 +102,11 @@ async function main() {
 				console.log(`\t-bf make|bob|imd|json\tCreate build files of a specific format`);
 				console.log(`\t\t\tExample: -bf make`);
 				console.log(``);
+				console.log(`\t-bl <name>\tSet the BIN_LIB environment variable`);
+				console.log(`\t\t\tbased on a user provided branch na,e`);
```
Comment (edmundreinhardt) [inline cli/src/index.ts:106]: sp `name` vs `na,me`

File: cli/src/index.ts [hunk 2]
```diff
@@ -96,6 +102,11 @@ async function main() {
...
+				console.log(`\t\t\tExample: -bl feature/123-cool-idea`);
```
Comment (edmundreinhardt) [inline cli/src/index.ts:107]: so in these examples, what does the library name become?

File: cli/src/builders/environment.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+
+export function getBranchLibraryName(currentBranch: string) {
+  const parts = branchSplit(currentBranch);
+  if (parts.length > 1) {
+    // We make A LOT of assumptions here about a valid branch name
```
Comment (edmundreinhardt) [inline cli/src/builders/environment.ts:5]: maybe document the assumed format?

File: cli/test/environment.test.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    expect(getBranchLibraryName(`feature/coolidea-thingy`)).toBe(`FEACOOLIDE`);
+  });
+
+  test('Type with description (longer, id at start)', () => {
+    expect(getBranchLibraryName(`feature/123-coolidea-thingy`)).toBe(`FEA123`);
+  });
+
+  test('Type with description (longer, id at end)', () => {
+    expect(getBranchLibraryName(`feature/coolidea-thingy-1234`)).toBe(`FEA1234`);
+  });
```
Comment (edmundreinhardt) [inline cli/test/environment.test.ts:35]: I don't understand why we get the tail end in this example and begining `coolide` everywhere else

File: cli/src/index.ts (hunk 2 above)
Comment (worksofliam) [reply cli/src/index.ts:107]: For `feature/123-cool-idea` it would be `FEA123`

File: cli/src/builders/environment.ts (hunk 3 above)
Comment (worksofliam) [reply cli/src/builders/environment.ts:5]: We just assume that devs use `/` and `-` in their branch names.

File: cli/test/environment.test.ts (hunk 4 above)
Comment (worksofliam) [reply cli/test/environment.test.ts:35]: Because it's more important we get a unique identifier like the digits than it is the text.

**Pull request merged**